tests/testthat/test-vivax.R

```diff
@@ -174,23 +176,123 @@ test_that('that vivax patent incidence rendering works', {
     renderer$render_mock(),
     1,
     'n_730_3650',
-    3,
+    2,
     timestep
   )
   
   mockery::expect_args(
     renderer$render_mock(),
     2,
     'n_detect_730_3650',
-    2,
+    1,
     timestep
   )
   
   mockery::expect_args(
     renderer$render_mock(),
     3,
     'p_detect_730_3650',
-    2,
+    1,
     timestep
+    )
+})
+
+
+test_that('Test age structure should not change vivax infectivity', {
+  
+  # Set all individuals to asymptomatic
+  # And ID immunity to not 0 (ID impacts age-specific asymptomatic infectivity)
+  parameters <- get_parameters(overrides = list(s_proportion = 0,
+                                                d_proportion = 0,
+                                                a_proportion = 1,
+                                                u_proportion = 0,
+                                                t_proportion = 0,
+                                                init_id  = 0.5))
+  
+  vivax_parameters <- get_parameters(parasite = "vivax",
+                                     overrides = list(s_proportion = 0,
+                                                      d_proportion = 0,
+                                                      a_proportion = 1,
+                                                      u_proportion = 0,
+                                                      t_proportion = 0,
+                                                      init_id  = 0.5))
+  
+  # Generate different age structure
+  year <- 365
+  ages <- round(c(0.083333, 1, 5, 10, 15, 20, 25, 30, 35, 40, 45,
+                  50, 55, 60, 65, 70, 75, 80, 85, 90, 95, 200) * year)
+  
+  deathrates <- rep(.1, length(ages)) / 365
+  
+  dem_parameters <- set_demography(
+    parameters,
+    agegroups = ages,
+    timesteps = 0,
+    deathrates = matrix(deathrates, nrow = 1)
   )
+  
+  vivax_dem_parameters <- set_demography(
+    vivax_parameters,
+    agegroups = ages,
+    timesteps = 0,
+    deathrates = matrix(deathrates, nrow = 1)
+  )
+  
+  # vivax asymptomatic infectivity should equal ca
+  expect_true(all(
+    c(create_variables(vivax_parameters)$infectivity$get_values(),
+      create_variables(vivax_dem_parameters)$infectivity$get_values())==vivax_parameters$ca))
+
+  # falciparum asymptomatic infectivity should not equal ca
+  expect_false(any(
+    c(create_variables(parameters)$infectivity$get_values(),
+      create_variables(dem_parameters)$infectivity$get_values())==vivax_parameters$ca))
+
+  # falciparum asymptomatic infectivity should change with age structure
+  expect_false(any(
+    c(create_variables(parameters)$infectivity$get_values() == create_variables(dem_parameters)$infectivity$get_values())))
+})
```

Comment:
I'm wondering if these checks for things not matching here are a little fragile? presumably with some set of reasonable parameters one or more of these values could `==ca` or could be the same with or without the age structure?

Would it be possible to simplify by
1. Initialising a population with 1 person at a given age
2. Running create variables, checking that the infectivity values are = to expected for both falicaparum and vivax
3. Change the age and repeat 2?

Reply:
Changed this test.